cranelift/codegen/src/isa/arm64/lower.rs

```diff
@@ -1505,31 +1505,35 @@ fn lower_insn_to_regs<C: LowerCtx<Inst>>(ctx: &mut C, insn: IRInst) {
             panic!("Should not appear: we handle BrTable directly");
         }
 
-        Opcode::Debugtrap | Opcode::Trap | Opcode::Trapif => {
-            let is_cond = op == Opcode::Trapif;
-            let maybe_trapinfo = inst_trapcode(ctx.data(insn)).map(|code| (ctx.srcloc(insn), code));
-
-            if is_cond {
-                let condcode = inst_condcode(ctx.data(insn)).unwrap();
-                let cond = lower_condcode(condcode);
-                let is_signed = condcode_is_signed(condcode);
-                // Verification ensures that the input is always a
-                // single-def ifcmp.
-                let ifcmp_insn = maybe_input_insn(ctx, inputs[0], Opcode::Ifcmp).unwrap();
-                lower_icmp_or_ifcmp_to_flags(ctx, ifcmp_insn, is_signed);
-
-                // Branch around the break instruction with inverted cond. Go straight
-                // to lowered one-target form; this is logically part of a single-in
-                // single-out template lowering.
-                let cond = cond.invert();
-                ctx.emit(Inst::CondBrLowered {
-                    target: BranchTarget::ResolvedOffset(8),
-                    kind: CondBrKind::Cond(cond),
-                });
-            }
-            ctx.emit(Inst::Brk {
-                trap_info: maybe_trapinfo,
+        Opcode::Debugtrap => {
+            ctx.emit(Inst::Brk);
+        }
+
+        Opcode::Trap => {
+            let trap_info = (ctx.srcloc(insn), inst_trapcode(ctx.data(insn)).unwrap());
+            ctx.emit(Inst::Udf { trap_info })
```

Comment:
Just checking -- does wasmtime continue to work properly with `udf` instead of `brk` for traps? (I suppose we'd need to test with a wasm program that explicitly traps somehow...) I seem to remember it just installs a bunch of signal handlers, including SIGTRAP and SIGILL, and if x86 generates SIGILL, it might Just Work. Just want to be sure!

Reply:
I'll check with a wasm test case that has unreachable in the start function.

Reply:
Confirmed it does work with the Spidermonkey unreachable test case.